feat: Add grounding convenience function (#378)
SAP/ai-sdk-js: packages/orchestration/README.md

### Grounding

Grounding enables integrating external, contextually relevant, domain-specific, or real-time data into AI processes.

> The grounding configuration can be provided as a raw JSON object or by using the `buildDocumentGroundingConfig()` function, function, which requires only the minimal mandatory values.

Author: ```suggestion
The grounding configuration can be provided as a raw JSON object or by using the `buildDocumentGroundingConfig()` function, which requires only the minimal mandatory values.
```

return response.getContent();
```

> `buildDocumentGroundingConfig` function provides convenience to create the document grounding service configuration using the minimal set of required values:

Author: ```suggestion
`buildDocumentGroundingConfig()` function provides convenience to create the document grounding service configuration using the minimal set of required values:
```

> `buildDocumentGroundingConfig()` function provides convenience to create the document grounding service configuration using the minimal set of required values:

Reviewer: ```suggestion
`buildDocumentGroundingConfig()` function provides convenience to create orchestration grounding configuration using the minimal set of required values:
```

```ts
import {
  buildDocumentGroundingConfig
} from '@sap-ai-sdk/orchestration';

const orchestrationConfig: {
  ...,

>   grounding: buildDocumentGroundingConfig(

Reviewer: Here, maybe `buildGroundingConfig`

Author: I named it so because the config `type` is `document_grounding_service` and this specific function only creates a configuration for this type. If in the future there is another type (donno if there will be), I imagine its name should be derived from that.

Author: Similar to how we say buildAzureFilter and not just buildFilter

Reviewer: I get your point. Still that feels  like we are configuring something for using document grounding service. I can understand why orchestration service named the type in this way, because they are going to call document grounding service. But from our user's side,  they are configuring orchestration grounding and should not be confused. @jjtang1985 What do you think.

Reviewer: Also, the type can be something else than `document_grounding_service`. So orchestration service by design can theoretically also utilizing other services besides document grounding service to do information retrieval.

Author: > Still that feels like we are configuring something for using document grounding service.

This is, in fact, exactly what the user does with the orchestration service though—they are aware of and configure the specific external provider. We also don't encapsulate/hide this for other modules (filters, masking).

> So orchestration service by design can theoretically also utilizing other services

You're right. The structure here differs from masking and filtering, where each `type` has its own config structure and therfore makes sense for different convenience functions for different providers. I'll check with Govind to confirm if this `config` is meant to be harmonized. Then we dont call this function documentGrounding

    input_params: ['groundingInput'],

>     output_param: 'groundingOutput',

Reviewer 2: [pp] I feel like this example isn't enough, as it leaves out how to configure the template, which isn't exactly obvious I'd say. Also, it's not intuitive what `'filter-id'` can/should be. If the ID is arbitrary, we may as well make it optional and create some ID for the user.

Author: The code example just above this (the Grounding section) has a detailed example (including templating). I just wanted to explain that the grounding part can be extracted to this function.

Reviewer 2: Ah okay, that should be sufficient. Maybe I'd switch the order though.. if users follow the instructions from top-down they only later learn that they could have used something simpler for most cases

>     filters: [{id: 'filter-id', data_repositories: ['repository-id'],}]

Reviewer: ```suggestion
    filters: [{ id: 'filter-id', data_repositories: ['repository-id'] }]
```
Not sure about the formatting

    ],
    defaults: {}
  },

>   grounding: buildDocumentGroundingConfig(

Reviewer 3: Same updates can be applied to the [sample app](https://github.com/SAP/ai-sdk-js/blob/main/sample-code/src/orchestration.ts#L232)

Author: Done
